### PR TITLE
Support secrets V2

### PR DESCRIPTION
**Pull request by MarinPostma**
This PR adds support for secrets V2. There are a couple of change in the payload that makes the secret version explicit, for a nicer handling with serde.

I have exported a sample secret from the actual backend for each version as test for compatibility with all versions of the secrets. The payloads in the tests are encrypted with a dummy RSA key.

### CODE REVIEW

**Comment by glommer** [comment]
looks good to me, skimming through.
@athoscouto ?

File: server/src/secrets.rs [hunk 1]
```diff
@@ -119,33 +127,48 @@ pub(crate) async fn get_secrets() -> Result<JsonObject> {
...
-    let data = match private_key {
-        None => data,
-        Some(private_key) => decrypt(&private_key, &data)?,
+    let secrets = match private_key {
+        None => serde_json::from_str(&data)?,
+        Some(private_key) => extract_secrets(&private_key, &data)?,
     };
-    Ok(serde_json::from_str(&data)?)
+
+    Ok(dbg!(secrets))
```
[the remark this answers is not in the document]
Comment (MarinPostma) [reply server/src/secrets.rs:135]: good catch!

**Comment by MarinPostma** [comment]
Please do not merge this PR right away!

**Comment by espindola** [comment]
> Please do not merge this PR right away!

You can mark it as draft for that.

**Comment by athoscouto** [comment]
@MarinPostma should we close this one?

**Comment by MarinPostma** [comment]
@athoscouto just waiting for us to complete the deployment, so we can always fallback to that solution if something unanticipated goes wrong.

**Pull request closed without merging**